src/code/z_actor.c

```diff
@@ -456,237 +469,242 @@ void Target_SetColors(TargetContext* targetCtx, Actor* actor, s32 type, PlayStat
-            gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_MODELVIEW | G_MTX_LOAD);
-            gSPDisplayList(POLY_XLU_DISP++, gZTargetArrowDL);
-        }
-
-        CLOSE_DISPS(play->state.gfxCtx);
+        gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, color->inner.r, color->inner.g, color->inner.b, 255);
+        gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_MODELVIEW | G_MTX_LOAD);
+        gSPDisplayList(POLY_XLU_DISP++, gZTargetArrowDL);
     }
+
+    CLOSE_DISPS(play->state.gfxCtx);
 }
 
 // OoT: func_8002C7BC
-void func_800B5814(TargetContext* targetCtx, Player* player, Actor* actor, GameState* gameState) {
-    PlayState* play = (PlayState*)gameState;
-    Actor* sp68 = NULL;
+void Target_Update(TargetContext* targetCtx, Player* player, Actor* targetedActor, PlayState* play) {
+    s32 pad;
+    Actor* actor = NULL;
     s32 category;
     Vec3f projectedPos;
     f32 invW;
 
+    // If no actor is currently targeted, then try to find a targetable actor
     if ((player->targetedActor != NULL) && (player->unk_AE3[player->unk_ADE] == 2)) {
-        targetCtx->unk_94 = NULL;
+        targetCtx->targetableOption = NULL;
     } else {
-        func_800BB8EC(gameState, &play->actorCtx, &sp68, &D_801ED920, player);
-        targetCtx->unk_94 = sp68;
+        Target_GetTargetActor(play, &play->actorCtx, &actor, &D_801ED920, player);
+        targetCtx->targetableOption = actor;
     }
 
-    if (targetCtx->unk8C != NULL) {
-        sp68 = targetCtx->unk8C;
-        targetCtx->unk8C = NULL;
-    } else if (actor != NULL) {
-        sp68 = actor;
+    if (targetCtx->nextTarget != NULL) {
+        actor = targetCtx->nextTarget;
+        targetCtx->nextTarget = NULL;
+    } else if (targetedActor != NULL) {
+        actor = targetedActor;
     }
 
-    if (sp68 != NULL) {
-        category = sp68->category;
+    if (actor != NULL) {
+        category = actor->category;
     } else {
         category = player->actor.category;
     }
 
-    if ((sp68 != targetCtx->arrowPointedActor) || (category != targetCtx->unk4A)) {
-        targetCtx->arrowPointedActor = sp68;
-        targetCtx->unk4A = category;
-        targetCtx->unk40 = 1.0f;
+    if ((actor != targetCtx->arrowPointedActor) || (category != targetCtx->arrowPointedActorCategory)) {
+        targetCtx->arrowPointedActor = actor;
+        targetCtx->arrowPointedActorCategory = category;
+        targetCtx->fairyMoveProgressFactor = 1.0f;
     }
 
-    if (sp68 == NULL) {
-        sp68 = &player->actor;
+    if (actor == NULL) {
+        actor = &player->actor;
     }
 
-    if (!Math_StepToF(&targetCtx->unk40, 0.0f, 0.25f)) {
-        f32 temp_f0;
-        f32 x;
-        f32 y;
-        f32 z;
-
-        temp_f0 = 0.25f / targetCtx->unk40;
+    if (!Math_StepToF(&targetCtx->fairyMoveProgressFactor, 0.0f, 0.25f)) {
+        f32 temp_f0 = 0.25f / targetCtx->fairyMoveProgressFactor;
```

Comment:
```suggestion
        f32 scale = 0.25f / targetCtx->fairyMoveProgressFactor;
```
or maybe `fairyMoveScale`?